src/parser/SparqlParser.cpp

```diff
@@ -338,81 +338,49 @@ void SparqlParser::parseWhere(ParsedQuery* query,
       currentPattern->_children.emplace_back(std::move(values));
       lexer_.accept(".");
     } else {
-      std::string subject;
-      if (lastSubject.empty()) {
-        if (lexer_.accept(SparqlToken::Type::VARIABLE)) {
-          subject = lexer_.current().raw;
-          query->registerVariableVisibleInQueryBody(lexer_.current().raw);
-        } else {
-          lexer_.expect(SparqlToken::Type::IRI);
-          subject = lexer_.current().raw;
+      // TODO cleanup
+      auto var = [](const Variable& var) { return var.name(); };
+      auto graphTerm = [](const GraphTerm& term) { return term.toSparql(); };
+      auto path = [](const PropertyPath& path) { return path.asString(); };
+      auto varOrTerm = [](VarOrTerm varOrTerm) {
+        if (Variable* var = std::get_if<Variable>(&varOrTerm)) {
+          return var->name();
+        } else if (GraphTerm* term = std::get_if<GraphTerm>(&varOrTerm)) {
+          return term->toSparql();
         }
-      } else {
-        subject = lastSubject;
-        lastSubject.clear();
-      }
+        AD_FAIL();  // Should be unreachable
+      };
```

Comment:
I did this this way because I got an error I didn't understand when using visit. According to this [Stackoverflow Post](https://stackoverflow.com/questions/63616709/incomplete-type-stdvariant-used-in-nested-name-specifier) using visit here would require some modifications to `VarOrTerm`. This is because `VarOrTerm` is not a `variant` directly but derives from `std::variant`.